lisa/tools/hugepages.py

```diff
@@ -0,0 +1,130 @@
+            if not matched_hugepage:
+                raise LisaException(
+                    f"no hugepage size found in {hugepage_size_dir_name}"
+                )
+            self._hugepage_sizes.add(matched_hugepage.group("hugepage_size_in_kb"))
+        return self._hugepage_sizes
+
+    def _enable_hugepages(self, enable_gibibyte_hugepages: bool = False) -> None:
+        echo = self.node.tools[Echo]
+
+        meminfo = self.node.tools[Free]
+        nics_count = len(self.node.nics.get_nic_names())
+
+        numa_nodes = self.node.tools[Lscpu].get_numa_node_count()
+        request_pages_2mb = (nics_count - 1) * 1024 * numa_nodes
+        request_pages_1gb = (nics_count - 1) * numa_nodes
+        memfree_2mb = meminfo.get_free_memory_mb()
+        memfree_1mb = meminfo.get_free_memory_gb()
+
+        # request 2iGB memory per nic, 1 of 2MiB pages and 1 GiB page
+        # check there is enough memory on the device first.
+        # default to enough for one nic if not enough is available
+        # this should be fine for tests on smaller SKUs
+
+        if enable_gibibyte_hugepages:
+            if memfree_1mb < (
+                request_pages_1gb * 2
+            ):  # account for 2MB pages by doubling ask
+                self.node.log.debug(
+                    "WARNING: Not enough 1GB pages available for DPDK! "
+                    f"Requesting {(request_pages_1gb * 2)} found {memfree_1mb} free. "
+                    "Test may fail if it cannot allocate memory."
+                )
+            request_pages_1gb = 1
+        else:
+            if memfree_2mb < request_pages_2mb:
+                self.node.log.debug(
+                    "WARNING: Not enough 2MB pages available for DPDK! "
+                    f"Requesting {request_pages_2mb} found {memfree_2mb} free. "
+                    "Test may fail if it cannot allocate memory."
+                )
+                request_pages_2mb = 1024
+
+        for i in range(numa_nodes):
+            if enable_gibibyte_hugepages:
+                echo.write_to_file(
+                    f"{request_pages_1gb}",
+                    self.node.get_pure_path(
+                        f"/sys/devices/system/node/node{i}/hugepages/"
+                        "hugepages-1048576kB/nr_hugepages"
+                    ),
+                    sudo=True,
+                )
+            else:
+                echo.write_to_file(
+                    f"{request_pages_2mb}",
+                    self.node.get_pure_path(
+                        f"/sys/devices/system/node/node{i}/hugepages/"
+                        "hugepages-2048kB/nr_hugepages"
+                    ),
+                    sudo=True,
+                )
+
+    def init_hugepages(self, enable_gibibyte_hugepages: bool = False) -> None:
+        mount = self.node.tools[Mount]
+        if enable_gibibyte_hugepages:
+            hugepage_sizes_in_kb = self.get_hugepage_sizes_in_kb()
+            if "1048576" not in hugepage_sizes_in_kb:
```

Comment:
A good start, but 'naked' string constants are a bad pattern. Maybe make an Enum and make get_hugepage_sizes return the Enum type instead of a string array.